src/webui/webapplication.cpp

```diff
@@ -431,15 +428,24 @@ void WebApplication::configure()
 
     const QString newLocale = pref->getLocale();
     if (m_currentLocale != newLocale) {
-        m_currentLocale = newLocale;
         m_translatedFiles.clear();
-        if (m_translator.load(m_rootFolder + QLatin1String("/translations/webui_") + m_currentLocale)) {
-            LogMsg(tr("Web UI translation for selected locale (%1) is successfully loaded.")
-                   .arg(m_currentLocale));
+
+        if (m_translator.load(m_rootFolder + QLatin1String("/translations/webui_") + newLocale)) {
+            m_currentLocale = newLocale;
+            LogMsg(tr("Web UI translation for selected locale (%1) has been successfully loaded.")
+                   .arg(newLocale));
         }
         else {
-            LogMsg(tr("Couldn't load Web UI translation for selected locale (%1). Falling back to default (en).")
-                   .arg(m_currentLocale), Log::WARNING);
+            if (isAltUIUsed) {
+                m_currentLocale = newLocale;
+                LogMsg(tr("Couldn't load Web UI translation for selected locale (%1).").arg(newLocale), Log::WARNING);
+            }
+            else {
+                m_currentLocale = QLatin1String("en");
+                m_translator.load(m_rootFolder + QLatin1String("/translations/webui_") + m_currentLocale);
```

Comment:
I'm sorry you confused me a little. I still don't see the point of having a translation for native language and load it here since it leads to the same result as if you don't do it at all.

Reply:
>load it here 

I suppose `m_translator` here should either be invalidated or load the correct file, since there is no invalid state for `QTranslator`, the other way is to load it anyway.

>I still don't see the point of having a translation for native language 

Otherwise you need to add condition checking for `locale == "en"` to suppress unwanted log message and other processing, which is a hack IMO and much less general way of doing it.

Reply:
> the other way is to load it anyway.

But why it's failed to load "en" translation currently? It was the cause of my attention to this problem.

Reply:
>But why it's failed to load "en" translation currently? It was the cause of my attention to this problem.

Now we only fall back to en when using built-in WebUI. In case of alt webui, it is always transitioned.

>Yes. I think it should be some file in web ui root containing name of default locale. 

I think this is over-engineering. Now only the log message is a bit of problem, the functionality is fine, at least from what I tested.